### PR TITLE
test: seed a completed lesson plan

### PR DESCRIPTION
**Pull request by codeincontext**
## Description

This PR adds an example for loading a seeded lesson plan at the start of a test

- Add "typical" lesson plan seeding, for a typical, completed lesson
- Add a test for just the download functionality of a completed lesson

## Screenshots

CleanShot 2024-09-25 at 18 04 24@2x

### CODE REVIEW

File: apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { bypassVercelProtection } from "../../helpers/vercel";
+import { isFinished } from "./helpers";
+
+test("Downloading a completed lesson plan", async ({ page }) => {
+  await test.step("Setup", async () => {
+    await clerkSetup();
+    await bypassVercelProtection(page);
+    await setupClerkTestingToken({ page });
+
+    await prepareUser(page, "typical");
```
Comment (codeincontext) [inline apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts:15]: This is the new shiny sauce!

File: apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    page.getByRole("heading", { name: "Download resources" });
+
+    // Click to download lesson plan
+    const downloadLessonPlan = page.getByTestId("chat-download-lesson-plan");
+    await downloadLessonPlan.click();
+
+    // Skip feedback form
+    if (await page.getByLabel("Skip").isVisible()) {
+      await page.getByLabel("Skip").click();
+    }
```
Comment (codeincontext) [inline apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts:35]: This part is extracted from the previous chat e2e test

File: packages/api/src/router/testSupport/seedChat/index.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+const fixtures = {
+  typical: typicalChatJson,
+};
+
+export const seedChat = async (
+  userId: string,
+  fixtureName: keyof typeof fixtures,
+) => {
+  const id = `e2e-typical-user${userId}`;
```
Comment (stefl) [inline packages/api/src/router/testSupport/seedChat/index.ts:13]: Should this have a dash before userId?

File: apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    // Generating
+    await expect(downloadLessonPlan).toContainText(
+      "Generating Lesson plan for export",
```
Comment (stefl) [inline apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts:41]: Maybe make this lowercase-compatible when we fix the case on this string?

File: apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    );
+
+    // Generated
+    await expect(downloadLessonPlan).toContainText(
+      "Download lesson plan (.docx)",
+      { timeout: 15000 },
+    );
+    await expect(downloadLessonPlan).toContainText(
+      "Download Lesson plan (.pdf)",
```
Comment (stefl) [inline apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts:50]: Same here - looks like incorrect casing on the page itself

**Review by stefl — APPROVED** [review]

File: apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts (hunk 4 above)
Comment (codeincontext) [reply apps/nextjs/tests-e2e/tests/aila-chat/downloads.test.ts:41]: Will do. Thanks

File: packages/api/src/router/testSupport/seedChat/index.ts (hunk 3 above)
Comment (codeincontext) [reply packages/api/src/router/testSupport/seedChat/index.ts:13]: I just had a look in my DB and this is actually generating IDs like `e2e-typical-useruser_2mZJKBdphP9dckutEjfwzcfmYei`. So I'm going to remove this prefix anyway. Thanks for the prompt to have a look

**Review by mantagen — APPROVED** [review]
Great!

**Pull request merged**